get_nam_data.pl

```diff
@@ -0,0 +1,497 @@
+#!/usr/bin/env perl
+#--------------------------------------------------------------
+# get_nam_data.pl: downloads background meteorology data from NCEP
+# for ASGS nowcasts and forecasts
+#--------------------------------------------------------------
+# Copyright(C) 2022 Jason Fleming
+#
+# This file is part of the ADCIRC Surge Guidance System (ASGS).
+#
+# The ASGS is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+#
+# ASGS is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with the ASGS.  If not, see <http://www.gnu.org/licenses/>.
+#--------------------------------------------------------------
+# ref: http://www.cpc.ncep.noaa.gov/products/wesley/fast_downloading_grib.html
+#--------------------------------------------------------------
+# sample lines to test this script:
+#
+## supply starting and ending nowcast cycle:
+#
+# perl get_nam_data.pl \
+#                 --startcycle 2005082900      \
+#                 --finishcycle 2005082906     \
+#                 --stage nowcast              \
+#
+# -- or --
+#
+## supply JSON file with a list of nowcast cycles to download:
+#
+# perl get_nam_data.pl \
+#                 --selectfile select_nam_nowcast.pl.json
+#                 --stage nowcast
+#
+# == or ==
+#
+## ask a specific forecast cycle:
+#
+# perl get_nam_data.pl \
+#                 --stage forecast \
+#                 --startcycle 2005082900      
+#--------------------------------------------------------------
+$^W++;
+use strict;
+use Net::FTP;
+use JSON::PP;
+use Getopt::Long;
+use Date::Calc;
+use File::Copy 'move';
+use File::Basename;
+use File::Path 'make_path';
+use Cwd;
+our $startcycle = "null"; # most recent cycle for which a nowcast was completed
+my $finishcycle = "null"; # nowcast end (meaningless in the forecast stage)
+my $selectfile = "null";  # array of cycles to download
+my $stage = "null";       # nowcast | forecast
+our $forecastcycle = "00,06,12,18";   # nam cycles to run a forecast
+my $backsite = "ftp.ncep.noaa.gov";   # ncep ftp site for nam data
+my $backdir = "/pub/data/nccf/com/nam/prod";    # dir on ncep ftp site
+my $localbasedir = cwd();    # main directory to download NAM data to
+our $scriptdir = dirname(__FILE__); # directory where this script is installed
+our $forecastlength = 84; # standard length of a NAM forecast
+our $max_retries = 20; # max number of times to attempt download of forecast file
+our $num_retries = 0;
+our $had_enough = 0;
```

Comment:
this is not a package so you can change the `our` to `my`

Reply:
I have removed all the "our" declarations. Not a big deal on `get_nam_data.pl`, but a lot more of an issue with `NAMtoOWIRamp.pl`. But we made it nonetheless.